components/nexus-repository/src/main/java/org/sonatype/nexus/repository/search/SearchFacet.java

```diff
@@ -14,33 +14,17 @@
 
 import org.sonatype.nexus.repository.Facet;
 import org.sonatype.nexus.repository.storage.Component;
+import org.sonatype.nexus.repository.storage.StorageTxPostCommitHook;
+
+import com.google.common.base.Supplier;
 
 /**
- * Search {@link Facet}, providing means to index/de-index component metadata.
+ * Search {@link Facet}, that index/de-index component metadata.
  *
  * @since 3.0
  */
 @Facet.Exposed
 public interface SearchFacet
-    extends Facet
+    extends Facet, Supplier<StorageTxPostCommitHook>
```

Comment:
Why is SearchFacet a Supplier<StorageTxPostCommitHook>? I would just register an anonymous implementation in StorageFacetImpl#doInit